### PR TITLE
[filter] Independency on order of the properties

### LINKED ISSUE #2650: [filter] Order of setting the properties
With the introduction of two versions of GstTensorFilterFramework, there has come up a bug with setting the properties.
The order of properties has now become important as setting certain properties depends on the version of the framework.
So, the version of framework is needed to set those properties, which creates a required order among some of the properties.

Given the two pipelines below:
```
pipelineA = "tensor_filter accelerator=true:GPU framework=tensorflow_lite"
pipelineB = "tensor_filter framework=tensorflow_lite accelerator=true:GPU"
```
Both the pipelines would have the correct framework. However, `accelerator` for `pipelineA` would be set to NULL because when `accelerator` property is set, `framework` is not yet set. This issue exists for properties `accelerator` and `is_updatable`. 

This is the inherent issue behind issue #2465

### PR DESCRIPTION
**Pull request by kparichay**
This patch adds independence on the order of setting the properties
Properties of tensor filter which depends on the framework can now be set before setting the framework as well
When setting the framework, these properties are set again from the cache

Resolves #2650

**Self evaluation:**
1. Build test: [x]Passed [ ]Failed [ ]Skipped
2. Run test: [x]Passed [ ]Failed [ ]Skipped

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

**Comment by kparichay** [comment]
V2:
Update assert to throw/catch in tensor_filter_support (needed by setting of cached properties)
Added checking of supported accelerators passed by the framework to be available at runtime before using them in tensor_filter

**Review by myungjoo — APPROVED** [review]
It's a bit difficult to review. (too many things in a single PR.) LGTM, anyway. (will need more reviewers than normal as this is a big fix)

**Comment by kparichay** [comment]
@myungjoo Noted. Will keep smaller and independent commits per PR in future. 
Modularization related to tensor filter accelerators will be done soon.

File: gst/nnstreamer/tensor_filter/tensor_filter_common.c [hunk 1]
```diff
@@ -1011,6 +1025,29 @@ _gtfc_setprop_FRAMEWORK (GstTensorFilterPrivate * priv,
...
+
+  /** set PROP_IS_UPDATABLE in case it was set before framework */
+  g_value_init (&val, G_TYPE_BOOLEAN);
+  g_value_set_boolean (&val, priv->is_updatable);
+  status = _gtfc_setprop_IS_UPDATABLE (priv, prop, &val);
+  if (status != 0) {
+    g_warning ("Set propery is-updatable failed with error: %d", status);
+    return status;
+  }
+  g_value_unset (&val);
```
Comment (dongju-chae) [inline gst/nnstreamer/tensor_filter/tensor_filter_common.c:1037]: How about moving `g_value_unset` to between line 1032 and 1033? Same issue in line 1048.

File: tests/nnstreamer_filter_tensorflow_lite/runTest.sh [hunk 2]
```diff
@@ -96,7 +96,7 @@ function run_pipeline() {
 }
 
 arch=$(uname -m)
-if [ "$arch" = "aarch64" ] || [ "$arch" = "armv7l" ]; then
+if [ "$arch" = "armv7l" ]; then
```
Comment (dongju-chae) [inline tests/nnstreamer_filter_tensorflow_lite/runTest.sh:99]: Why aarch64 is disabled?

File: gst/nnstreamer/tensor_filter/tensor_filter_common.c [hunk 3]
```diff
@@ -24,6 +24,14 @@
...
 
 #include <string.h>
+#if defined(__aarch64__) || defined(__arm__)
+#include <sys/auxv.h>
+#if defined(__ANDROID__)
+#include <asm/hwcap.h>
+#else /* __ANDROID __ */
+#include <asm-arm/hwcap.h>
+#endif /* __android __ */
+#endif /* __arch64__ __arm__ */
```
Comment (dongju-chae) [inline gst/nnstreamer/tensor_filter/tensor_filter_common.c:34]: IMO, it's better to create an additional header file like this [framework](https://ncnn.docsforge.com/master/api/ncnn/cpu_support_arm_neon/) in another PR,

Comment (kparichay) [reply gst/nnstreamer/tensor_filter/tensor_filter_common.c:34]: Yes, the accelerator related stuff will move to different files.

File: gst/nnstreamer/tensor_filter/tensor_filter_common.c (hunk 1 above)
Comment (kparichay) [reply gst/nnstreamer/tensor_filter/tensor_filter_common.c]: Sorry for the late reply. this is done.

File: tests/nnstreamer_filter_tensorflow_lite/runTest.sh (hunk 2 above)
Comment (kparichay) [reply tests/nnstreamer_filter_tensorflow_lite/runTest.sh]: This was wrong change. Has been reverted.

**Comment by kparichay** [comment]
@dongju-chae Sorry for the late update. I have updated as per your comments. Please review.

File: tests/nnstreamer_filter_tensorflow_lite/runTest.sh [hunk 4]
```diff
@@ -96,7 +96,7 @@ function run_pipeline() {
 }
 
 arch=$(uname -m)
-if [ "$arch" = "aarch64" ] || [ "$arch" = "armv7l" ]; then
+if ["$arch" = "aarch64"] || [ "$arch" = "armv7l" ]; then
```
Comment (dongju-chae) [inline tests/nnstreamer_filter_tensorflow_lite/runTest.sh:99]: Mistake here?

Comment (kparichay) [reply tests/nnstreamer_filter_tensorflow_lite/runTest.sh]: yes, fixed.

**Review by dongju-chae — APPROVED** [review]
LGTM

**Pull request merged**